load_testing_locust.md

```diff
@@ -0,0 +1,592 @@
+
+
+Take note of the Client ID and Client Secret.
+
+14. Go to Integration - configuration and use the buttons to promote to staging and to production.
+
+<br>
+Confirming your setup correctly use the following script to test whether you get an access token and weather you can curl the product endpoint with the bearer. 
+
+<br>
+
+
+```
+#!/bin/bash
+# user-sso route
+KCHOST=https://keycloak-redhat-rhoam-user-sso.apps.fwaters-ccs2.xqpg.s2.devshift.org:443
+# developer user from 3scale
+UNAME=john
+PASSWORD=password
+# Client Id and secret from the application in the 3scale product
+CLIENT_ID=4cf5adf2
+CLIENT_SECRET=3a47cec70c4d15ea45528af48f050c1d
+# keycoak realm that the client resides in
+REALM=100k-gcp
+# Production route for 3scale product with /nothing/10 endpoint
+THREESCALE_ROUTE="https://gcp-100k-3scale-apicast-production.apps.fwaters-ccs2.xqpg.s2.devshift.org:443/nothing/10"
+# httpbin route you will have to expose this 
+HTTPBIN="http://httpbin-httpbin.apps.fwaters-ccs2.xqpg.s2.devshift.org/nothing/10"
+
+ACCESS_TOKEN=`curl \
+  -d "client_id=$CLIENT_ID" -d "client_secret=$CLIENT_SECRET" \
+  -d "username=$UNAME" -d "password=$PASSWORD" \
+  -d "grant_type=password" \
+  "$KCHOST/auth/realms/$REALM/protocol/openid-connect/token"  | jq -r '.access_token'`
+echo $ACCESS_TOKEN
+echo THREESCALE_ROUTE ===============================================================
+curl -X POST -H "Authorization: Bearer $ACCESS_TOKEN" $THREESCALE_ROUTE -d '{"login":"my_login","password":"my_password"}'
+echo HTTPBIN ==================================================================
+curl -X POST $HTTPBIN -d '{"login":"my_login","password":"my_password"}'
+```
+
+
+Once this is confirmed you can populate the required auth file. See Auth section below.
+
+<br>
+
+## Setting up locust
+
+Locust is a load testing tool that can be run locally or on a container on another cluster
+
+* Fork repo - [https://github.com/integr8ly/locust-integreatly-operator](https://github.com/integr8ly/locust-integreatly-operator)
+* Clone it locally
+* Install locust
+
+```
+pip3 install locust 
+```
+
+**Config**
+
+constant_pacing is used to get to the correct number of requests per second(rps) for the load testing on locust. It also takes into account the number of users. It is the inverse of constant_throughput. In order to increase the rps you need to decrease the constant pacing value. When changing this be sure to only update in small increments. More info on this function can be found [here](https://docs.locust.io/en/stable/api.html#locust.wait_time.constant_pacing).
+
+
+* 1.157 rps pacing value 9.25 for 100k at 11 users
+* 11.57 rps pacing value 1.25 for 1m at 14 users
+* 57.87 rps pacing value 0.207 for 5m at 12 users
+* 110.00 rps pacing value 0.09 for 10m at 10 user
+* 231.48 rps pacing value * for 20m at 12 user
+* ~550 rps pacing value 0.01 for 50m at 32 users
+* 1157.41 rps pacing value * for 100m at 12 user
```

Comment:
We should go with the constant pacing of 1 and set the number of users to just above the rate , Only one we would have to play around would 100k as it would be lower